api/v1/storageprofile_types.go

```diff
@@ -92,3 +97,13 @@ const (
 func init() {
 	SchemeBuilder.Register(&StorageProfile{}, &StorageProfileList{})
 }
+
+func (sp *StorageProfile) GetSpecHash() string {
```

Comment:
correct me if I'm missing something, why can't we just use the UUID since all the fields of SP are static once set?

Reply:
@leela UUID locks you into a specific CRD instance. 
What happens if, for one reason or another, we need to delete one instance and recreate it?
What happens if we are to back up and restore a cluster? 
What happens if we have 2 SPs which have the exact same spec? should we use 2 different CBPs or Just one?

Relying on UUIDs or resource names is problematic. With hashes, we get to relay on the configuration itself and not on an identity.

Reply:
got it, thanks. Btw, you pinged wrong handle, `s/leela/leelavg/`